ts/src/gate.ts

```diff
@@ -4581,16 +4688,27 @@ export default class gate extends Exchange {
                 }
             }
         }
-        const [ type, query ] = this.handleMarketTypeAndParams ('fetchPositions', market, params);
-        if (type !== 'swap' && type !== 'future') {
-            throw new ArgumentsRequired (this.id + ' fetchPositions requires a type parameter, "swap" or "future"');
+        let type = undefined;
+        let request = {};
+        [ type, params ] = this.handleMarketTypeAndParams ('fetchPositions', market, params);
```

Comment:
@Dan-krm whenever we have a symbol available we should use that to infer the `type`

Reply:
Okay, I'm not sure why I did this, I think it felt cleaner to do it this way and it would still be inferred through the market but I'll change it

Reply:
@Dan-krm actually my bad, seeing only a small part of the code confused me, if we were inferring using market it was ok, sorry

Reply:
I mean we were inferring using market by defining market in handleMarketTypeAndParams, I think it's okay now using market['type'] instead, let me know if I should change it

Reply:
I removed handleMarketTypeAndParams in fetchPosition but I left it in fetchPositions where this comment was made since fetchPositions can be called with the symbol being undefined